tests/test_batchupload.py

```diff
@@ -67,12 +67,12 @@ def test_cancel(server):
     batch.delete(0)
 
 
-def test_data():
+def test_data(tmp_path):
     blob = BufferBlob(data="data", name="Test.txt", mimetype="text/plain")
     with blob:
         assert blob.data
 
-    test = "test_file"
+    test = os.path.join(str(tmp_path), "test_file")
```

Comment:
Outch! This is not attractive :D
Just use a simple `Path`:
```suggestion
    file_out = tmp_path / "test_file"
```
And later, use `str(file_out)` in operations and `file_out.write_bytes(...)` instead of `with open(file_out...)`.